stuff/cron/build_problem_rec_model.py

```diff
@@ -0,0 +1,333 @@
+#!/usr/bin/python3
+
+'''Builds the user recommendation model.
+
+This script reads user submissions and builds a recommendation model for
+which problems to attempt after a user has just sucessfully solved a problem.
+
+The model is a probability distribution of which problems a random user would
+attempt to solve next, based solely on the sequence of recently solved problems
+up to the latest one.
+'''
+
+import argparse
+import logging
+import sqlite3
+import warnings
+
+from typing import List, Tuple
+
+import MySQLdb.connections
+
+import numpy as np
+import pandas as pd
+
+import lib.db
+import lib.logs
+
+
+# Default training parameters.
+_TRAIN_FRACTION = 0.8
+_NUM_FOLLOWUPS = 3
+_FOLLOWUP_DECAY = 0.4
+
+
+# Type aliases for type checking
+ProblemList = List[int]
+
+
+def mean_average_precision(predicted: ProblemList,
+                           expected: ProblemList,
+                           k: int) -> float:
```

Comment:
como estás regresando None in L44, puedes hacer esto

```suggestion
                           k: int) -> Optional[float]:
```

?

Reply:
Done